cli/popper/template.py

```diff
@@ -0,0 +1,143 @@
+import os
+from popper import utils
+
+class ReadMe:
+    def __init__(self):
+        self.repo_name = utils.get_repo_name()
+        self.project_root = utils.get_project_root()
+
+    def write_readme(self, content, path):
+        """ Creates a README.md file with the specified content
+        at the given path.
+
+        Args:
+            content (str): The contents for the README.md file
+            path (str): The absolute path where the README.md file
+                        is to be created,
+        """
+
+        with open(os.path.join(path, 'README.md'), 'w') as f:
+            f.write(content)
+
+    def init_project(self):
+        content = """
+# {0}
+
+This repository contains [Popper](https://github.com/systemslab/popper) pipelines. To show a list of available pipelines using the [`popper` CLI tool](https://github.com/systemslab/popper):
+
+```bash
+cd {0}
+popper ls
+```
+
+to execute one of the pipelines:
+
+```bash
+popper run <pipeline-name>
+```
+
+where `<pipeline-name>` is one of the pipelines in the repository. For more on what other information from this repository is available, you can run:
+
+```bash
+popper --help
+```
+"""
+        content = content.format(self.repo_name)
+        self.write_readme(content, self.project_root)
+
+    def init_pipeline(self, pipeline_path, stages, envs):
+        """ Generates a README template for the newly initialized
+        pipeline.
+
+        Args:
+            pipeline_path (str): The absolute path of the pipeline.
+            stages (str): Contains all the stages of the pipeline separated
+                          by comma.
+            envs (list): Contains a list of the environments on which the
+                         pipeline can be executed.
+
+        """
+
+        pipeline_name = pipeline_path.split('/')[-1]
+        content = """
+# `{}`
+<!--
+NOTE: replace all the **TODO** marks with your own content.
+-->
+"""
+        content = content.format(pipeline_name)
+        if (len(stages)) > 0:
+            content += """
+**TODO**: insert high-level description of the pipeline. It consists of the following stages:
+"""
+            for i, stage in enumerate(stages.split(',')):
+                content += """
+* [`{0}`](./{0}.sh). **TODO**. Add high-level description of stage `{0}`.
+"""
+                content = content.format(stage)
+
+        content += """
+
+# Obtaining the pipeline
+
+To add this pipeline to  the [`popper` CLI tool](https://github.com/systemslab/popper):
+
+```bash
+cd your-repo
+popper add org/{0}/{1}
+```
+
+**TODO**: replace `org` appropriately.
```

Comment:
We could try to get the `org` name using git first? And if that fails, then we add the `**TODO**` node.